presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java

```diff
@@ -0,0 +1,346 @@
+        assertQuery("SELECT ST_Contains(ST_GeometryFromText('Polygon((0 2,1 1,0 -1,0 2))'), ST_GeometryFromText('Polygon((-1 3,2 1,0 -3,-1 3))'))", "SELECT false");
+        assertQuery("SELECT ST_Contains(ST_GeometryFromText('LineString(20 20,30 30)'), ST_GeometryFromText('Point(25 25)'))", "SELECT true");
+        assertQuery("SELECT ST_Contains(ST_GeometryFromText('LineString(20 20,30 30)'), ST_GeometryFromText('Point(25 25)'))", "SELECT true");
+    }
+
+    @Test
+    public void testSTCrosses()
+            throws Exception
+    {
+        assertQuery("select ST_Crosses(ST_GeometryFromText('LineString(0 0, 1 1)'), ST_GeometryFromText('LineString(1 0, 0 1)'))", "select true");
+        assertQuery("select ST_Crosses(ST_GeometryFromText('Polygon ((1 1, 1 4, 4 4, 4 1))'), ST_GeometryFromText('Polygon ((2 2, 2 5, 5 5, 5 2))'))", "select false");
+        assertQuery("select ST_Crosses(ST_GeometryFromText('Polygon ((1 1, 1 4, 4 4, 4 1))'), ST_GeometryFromText('LineString(2 0, 2 3)'))", "select true");
+        assertQuery("select ST_Crosses(ST_GeometryFromText('LineString(0 0, 1 1)'), ST_GeometryFromText('LineString(1 0, 0 1)'))", "select true");
+    }
+
+    @Test
+    public void testSTDisjoint()
+            throws Exception
+    {
+        assertQuery("select ST_Disjoint(ST_GeometryFromText('LineString(0 0, 0 1)'), ST_GeometryFromText('LineString(1 1, 1 0)'))", "select true");
+        assertQuery("select ST_Disjoint(ST_GeometryFromText('Polygon ((1 1, 1 3, 3 3, 3 1))'), ST_GeometryFromText('Polygon ((4 4, 4 5, 5 5, 5 4))'))", "select true");
+        assertQuery("select ST_Disjoint(ST_GeometryFromText('Polygon ((1 1, 1 3, 3 3, 3 1))'), ST_GeometryFromText('Polygon ((4 4, 4 5, 5 5, 5 4))'))", "select true");
+    }
+
+    @Test
+    public void testSTEquals()
+            throws Exception
+    {
+        assertQuery("select ST_Equals(ST_GeometryFromText('LineString(0 0, 2 2)'), ST_GeometryFromText('LineString(0 0, 2 2)'))", "select true");
+        assertQuery("select ST_Equals(ST_GeometryFromText('LineString(0 0, 2 2)'), ST_GeometryFromText('LineString(0 0, 2 2)'))", "select true");
+    }
+
+    @Test
+    public void testSTIntersects()
+            throws Exception
+    {
+        assertQuery("select ST_Intersects(ST_GeometryFromText('LineString(8 4, 4 8)'), ST_GeometryFromText('Polygon ((2 2, 2 6, 6 6, 6 2))'))", "select true");
+        assertQuery("select ST_Intersects(ST_GeometryFromText('LineString(8 4, 4 8)'), ST_GeometryFromText('Polygon ((2 2, 2 6, 6 6, 6 2))'))", "select true");
+        assertQuery("select ST_Intersects(ST_GeometryFromText('LineString(8 4, 4 8)'), ST_GeometryFromText('Polygon ((2 2, 2 6, 6 6, 6 2))'))", "select true");
+        assertQuery("select ST_Intersects(ST_GeometryFromText('LineString(8 4, 4 8)'), ST_GeometryFromText('Polygon ((2 2, 2 6, 6 6, 6 2))'))", "select true");
+    }
+
+    @Test
+    public void testSTOverlaps()
```

Comment:
this function as well as ST_Relate, ST_Touches and ST_Within need beefier coverage